test/datastore/error_matchers.dart

```diff
@@ -9,39 +9,32 @@ import 'dart:io';
 import 'package:test/test.dart';
 import 'package:gcloud/datastore.dart';
 
-class _ApplicationError extends TypeMatcher {
-  const _ApplicationError() : super("ApplicationError");
-  bool matches(item, Map matchState) => item is ApplicationError;
+class _ApplicationError extends TypeMatcher<ApplicationError> {
```

Comment:
Actually, I think we can get rid of the classes entirely now:

```dart
const isApplicationError = const TypeMatcher<ApplicationError>()
```

Reply:
Agreed.

Reply:
Implemented.